launch_ros/launch_ros/actions/node.py

```diff
@@ -177,6 +183,78 @@ def __init__(
 
         self.__logger = launch.logging.get_logger(__name__)
 
+    @staticmethod
+    def parse_nested_parameters(params, parser):
+        """Normalize parameters as expected by Node constructor argument."""
+        def get_nested_dictionary_from_nested_key_value_pairs(params):
+            """Convert nested params in a nested dictionary."""
+            # TODO(ivanpauno): If our schema checking is not powerfull enough
+            # this could easily end in an infinite loop.
+            # In that case, we should do some extra formal checking before processing.
+            param_dict = {}
+            for param in params:
+                name = param.get_attr('name')
+                value = param.get_attr('value', types='guess', optional=True)
+                if value is not None:
```

Comment:
This logic is a bit confusing, but I don't have a suggestion on how to make it clearer off-hand. Like if 'guess` returns `None` then implicitly it is a list of entities?

Reply:
Yes this is a little confusing.
You have two mutually exclusive options:
- parameter specifying `name` and `value`.
- parameter specifying `name`, and having some other nested parameter entities.

Only having a `name` attribute and nothing more is an error (it's not clear in the code above, I will update).

The first options looks in an xml like:
```xml
<param name="asd" value="1"/>
```

and the second as:
```xml
<param name="asd">
    <param name="bsd" value="1"/>
</param>
```

The same rule is applied recursively for nested parameter entities.

It's also a third option, that can only be applied in a "root" param tag. It looks like:
```xml
<param from="path/to/file"/>
```
This option can not have other entities as children.

I haven't added tests for that option. I will do that before merging.